packages/neuron-wallet/src/services/wallets.ts

```diff
@@ -45,17 +48,49 @@ export class FileKeystoreWallet implements Wallet {
       throw new IsRequired('Name')
     }
 
+    if (!extendedKey && !device) {
+      throw new IsRequired('Extended Public Key or Device Info')
+    }
+
     this.id = id
     this.name = name
     this.extendedKey = extendedKey
+    this.device = device
   }
 
-  accountExtendedPublicKey = (): AccountExtendedPublicKey => {
-    return AccountExtendedPublicKey.parse(this.extendedKey) as AccountExtendedPublicKey
+  public toJSON = () => ({
+    id: this.id,
+    name: this.name,
+    extendedKey: this.extendedKey,
+    device: this.device
+  })
+
+  public fromJSON = () => {
+    throw new Error('not implemented')
   }
 
-  static fromJSON = (json: WalletProperties) => {
-    return new FileKeystoreWallet(json)
+  public isHardware = (): boolean => {
+    throw new Error('not implemented')
+  }
```

Comment:
TypeScript's `abstract class` can omit the implementation of functions like interface:
```typescript
public abstract isHardware (): boolean
```
Should be enough.